.github/workflows/build-and-deploy.yml

```diff
@@ -57,23 +51,36 @@ jobs:
           cache-to: type=inline
           context: ./cpp/docker
       
-      # Build site
+      # Generate wasm files
       - name: Run wasm setup
         working-directory: ./cpp/
         run: ./build-compiler-in-docker.sh ${{ env.DOCKER_HUB_NAMESPACE }}/${{ env.DOCKER_REPO }}:${{ github.sha }}
 
+      - name: Upload artifacts to B2
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: ./scripts/upload_to_b2.sh mlir-playground ${{ secrets.B2_APPLICATION_KEY_ID }} ${{ secrets.B2_APPLICATION_KEY }} ./public/wasm files
+      
+      - name: Remove wasm files from production build
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: rm -r ./public/wasm
+
+      # Build site
       - name: Install npm dependencies
         run: npm install
+
+      - name: Set build environment
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: echo "NODE_ENV=production" >> $GITHUB_ENV
       
       - name: Run npm build
         run: npm run build
       
       - name: Export static website
         run: npx next export
       
-      # Push docker image if this is run on `main`
+      # Push docker image if on `main`
       - name: Push Docker image
-        if: ${{ github.ref == 'refs/heads/main' }}
+        if: ${{ github.ref == 'refs/heads/main' }}d
```

Comment:
is this a typo? @zyx-billy

Reply:
oh 🤦 yeah good find, looks like a typo, let's remove it. Interestingly it didn't really affect functionality...